pkg/export/setup/setup.go

```diff
@@ -230,3 +238,73 @@ func loadKubeConfig(kubeconfigPath string) (*rest.Config, error) {
 func ExtraArgs() ([]string, error) {
 	return shlex.Split(os.Getenv(ExtraArgsEnvvar))
 }
+
+// tryPopulateUnspecifiedFromMetadata assumes we are in GCP, with unknown state
+// of the metadata server. This is best-effort, so when the metadata server
+// is not accessible on GCP, because it was disabled (404 errors), or not accessible
+// (connection refused, slow network, e.g. sandbox + metadata disabled)
+// it's a noop. Make sure to pass context with a timeout.
```

Comment:
suggestion: Could we add a TODO here referencing we can update/slim this code down once  https://github.com/googleapis/google-cloud-go/pull/10370 is merged?